models/products/scratch/snowflake/snowplow_ecommerce_product_interactions_this_run.sql

```diff
@@ -0,0 +1,115 @@
+{{
+  config(
+    tags=["this_run"],
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with product_info as (
+  select
+    {{ dbt_utils.surrogate_key(['t.event_id', 'r.value:id']) }} as product_event_id,
+    t.event_id,
+    t.page_view_id,
+
+    -- session fields
+    t.domain_sessionid,
+    t.page_view_in_session_index,
+
+    -- user fields
+    t.domain_userid,
+    t.network_userid,
+    t.user_id,
+    t.ecommerce_user_id,
+
+    -- timestamp fields
+    t.derived_tstamp,
+    DATE(derived_tstamp) as derived_tstamp_date,
+
+    -- ecommerce product fields
+    r.value:id::string as product_id,
+    r.value:category::string as product_category,
+    SPLIT(r.value:category, '{{ var("snowplow__categories_separator", "/") }}') as product_categories_split,
+    r.value:currency::string as product_currency,
+    r.value:price::float as product_price,
+    r.value:brand::string as product_brand,
+    r.value:creative_id::string as product_creative_id,
+    r.value:inventory_status::string as product_inventory_status,
+    r.value:list_price::float as product_list_price,
+    r.value:name::string as product_name,
+    r.value:position::integer as product_list_position,
+    r.value:quantity::integer as product_quantity,
+    r.value:size::string as product_size,
+    r.value:variant::string as product_variant,
+    t.ecommerce_action_type,
+    t.ecommerce_action_name,
```

Comment:
Should we carry through the decimal decision here?

Reply:
Yep, good catch!